simpeg/optimization.py

```diff
@@ -821,7 +835,11 @@ def projection(self, x):
         Make sure we are feasible.
 
         """
-        return np.median(np.c_[self.lower, x, self.upper], axis=1)
+        if self.lower is not None:
+            x = np.maximum(x, self.lower)
+        if self.upper is not None:
+            x = np.minimum(x, self.upper)
+        return x
```

Comment:
`lower` and `upper` are now both optional arguments (even though they are defaulted to `-np.inf` and `np.inf` in the concrete implementations, they could be set to `None` for the same effect and avoid an operation.